internal/db/schema/migrations/oss/postgres/101/01_app_token_base.up.sql

```diff
@@ -0,0 +1,246 @@
+-- Copyright (c) HashiCorp, Inc.
+-- SPDX-License-Identifier: BUSL-1.1
+
+begin;
+
+
+  -- Create the enumeration table for app token global grant scope
+  create table app_token_global_grant_scope_enm (
+    name text primary key
+      constraint only_predefined_app_token_global_grant_scope_allowed
+        check(
+          name in (
+            'individual',
+            'children',
+            'descendants'
+          )
+        )
+  );
```

Comment:
Why create a table & FK constraint for these enums instead of creating an enum type? e.g.

```sql
create type app_token_global_grant_scope_enum as enum ('individual', 'children', 'descendants');
```

By using an enum type, we'd have one less table to `join` on